pkg/config/v2/config.go

```diff
@@ -0,0 +1,79 @@
+// @license
+// Copyright 2021 Dynatrace LLC
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v2
+
+import (
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/config/v2/coordinate"
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/config/v2/parameter"
+	envParam "github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/config/v2/parameter/environment"
+	refParam "github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/config/v2/parameter/reference"
+	valueParam "github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/config/v2/parameter/value"
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/config/v2/template"
+)
+
+const (
+	// id is a special parameter. it is not allowed to be set via the config,
+	// but needs to work as normal parameter otherwise (e.g. it can be referenced).
+	ID_PARAMETER = "id"
+
+	// the name parameter is special in that it needs to exist for a config.
+	NAME_PARAMETER = "name"
+)
+
+// map defining parameter name to parameter
+type Parameters map[string]parameter.Parameter
+
+// struct defining a configuration which can be deployed.
+type Config struct {
+	// template used to render the request send to the dynatrace api
+	Template template.Template
+	// coordinates which specify the location of this configuration
+	Coordinate coordinate.Coordinate
+	// group this config belongs to
+	Group string
+	// name of the environment this configuration is for
+	Environment string
+	// map of all parameters which will be resolved and are then available
+	// in the template
+	Parameters Parameters
+	// slice of references as specified by the paramters. needed so that the
+	// configurations can be sorted in before the deployment phase.
+	References []coordinate.Coordinate
+	// flag inidicating if the deployment of this configuration should be skipped
+	Skip bool
+}
+
+// used to determine if this config represents the given coordinates
+func (c *Config) MatchReference(ref coordinate.Coordinate) bool {
+	return c.Coordinate.Match(ref)
+}
+
+// test if this config has a dependency on the given config.
```

Comment:
Same for method comments, they should start with method name
`// HasDependencyOn`